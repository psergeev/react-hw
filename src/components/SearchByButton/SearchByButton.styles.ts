const styles: any = {
    button: {
        marginLeft: '10px',
        padding: '5px 15px',
        color: 'white',
        textDecoration: 'none',
        backgroundColor: '#3C3C3B',
        textTransform: 'uppercase',
    },

    selected: {
        backgroundColor: '#F45263 !important'
    }
};

export default styles;
