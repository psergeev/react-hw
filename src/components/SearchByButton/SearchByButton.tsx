import { Button, WithStyles, withStyles } from '@material-ui/core';
import * as React from 'react';
import styles from './SearchByButton.styles';

interface Props {
    value: string;
    selected: boolean;
    handleSearchByClick: (value: string) => void;
}

type PropsWithStyles = Props & WithStyles<'button' | 'selected'>;

function SearchByButton({ value, selected, handleSearchByClick, classes }: PropsWithStyles) {
    const onClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
        event.preventDefault();

        return handleSearchByClick(value);
    };

    return (
        <Button
            onClick={onClick}
            color="primary"
            className={`${classes.button} ${(selected ? classes.selected : '')}`}
        >
            {value}
        </Button>
    );
}

export default React.memo(withStyles(styles)(SearchByButton));
