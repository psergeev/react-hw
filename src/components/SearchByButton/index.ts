import { withStyles } from '@material-ui/core';
import SearchByButton from './SearchByButton';
import styles from './SearchByButton.styles';

export default withStyles(styles)(SearchByButton);
