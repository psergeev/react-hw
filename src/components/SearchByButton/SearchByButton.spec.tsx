import classes from 'identity-obj-proxy';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SearchByButton from './SearchByButton';

test('SearchByButton component works as expected', () => {
    const component = renderer.create(
        <SearchByButton
            classes={classes}
            value="value"
            selected={false}
            handleSearchByClick={jest.fn()}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('SearchByButton component can be selected', () => {
    const component = renderer.create(
        <SearchByButton
            classes={classes}
            value="value"
            selected
            handleSearchByClick={jest.fn()}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
