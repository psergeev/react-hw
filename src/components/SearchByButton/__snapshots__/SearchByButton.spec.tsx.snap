// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByButton component can be selected 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-textPrimary-6 MuiButton-flat-8 MuiButton-flatPrimary-9 SearchByButton-button-1 SearchByButton-selected-2"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    value
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`SearchByButton component works as expected 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-textPrimary-6 MuiButton-flat-8 MuiButton-flatPrimary-9 SearchByButton-button-1 "
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    value
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;
